### PR TITLE
Additional services

### PR DESCRIPTION
**Pull request by jonwinton**
Exposing some additional kiln services

### CODE REVIEW

File: services/components/add-component-handler.js [hunk 1]
```diff
@@ -133,6 +133,8 @@ function addHandler(button, options, prevRef) {
 
 module.exports = addHandler;
 module.exports.getParentEditableElement = getParentEditableElement;
+_.set(window, 'kiln.services.getAddableComponents', getAddableComponents); // export for plugins
```
Comment (nelsonpecora) [inline services/components/add-component-handler.js]: why not export the `exports`?

Comment (jonwinton) [inline services/components/add-component-handler.js]: Hmm I haven't needed `getParentEditableElement` so I didn't want to expose something that might never be used. But if we want to expose it I'm down.

File: services/edit/index.js [hunk 2]
```diff
@@ -650,6 +658,29 @@ function getComponentRef(uri) {
   }
 }
 
+/**
+ * get the components HTML and pass along any query paramaters
```
Comment (nelsonpecora) [inline services/edit/index.js]: component's, parameters

File: services/edit/index.js [hunk 3]
```diff
@@ -650,6 +658,29 @@ function getComponentRef(uri) {
...
+ * that may need to be exposed to the component in it's server.js
```
Comment (nelsonpecora) [inline services/edit/index.js]: its

File: services/edit/index.js [hunk 4]
```diff
@@ -650,6 +658,29 @@ function getComponentRef(uri) {
...
+ * file. This is handy for passing the current page URL to the
+ * component for re-rendering when in edit mode.
+ *
+ * TODO: Currently this is ONLY used for the Space component because
+ * querying for artile tags is limited. There is a pending update
```
Comment (nelsonpecora) [inline services/edit/index.js]: article

File: services/edit/db.js [hunk 5]
```diff
@@ -247,6 +247,17 @@ function getHTML(uri) {
 
 /**
  * @param {string} uri
+ * @param {string} queries
+ * @return {Promise}
+ */
+function getHTMLWithQuery(uri, queries) {
+  var queryUrl = uri + extHtml + editMode + queries;
```
Comment (nelsonpecora) [inline services/edit/db.js]: queries should probably be an object, `_.assign`'d to `{ edit: true}`, then stringified into `?edit=true&query1=whatever&query2=whatever` so consumers don't need to care about the ampersands/question marks

**Comment by nelsonpecora** [comment]
👍  when tests pass

**Pull request merged**